tests/metrics_publishing.rs

```diff
@@ -156,126 +131,144 @@ async fn f64_histogram_is_exported() {
-                );
+        let mut rm = ResourceMetrics {
+            resource: Resource::default(),
+            scope_metrics: Vec::new(),
+        };
+        self.reader.collect(&mut rm)?;
+
+        assert!(!rm.scope_metrics.is_empty());
+
+        rm.scope_metrics.into_iter().for_each(|scope_metrics| {
+            assert_eq!(scope_metrics.scope.name, INSTRUMENTATION_LIBRARY_NAME);
+            assert_eq!(
+                scope_metrics.scope.version.unwrap().as_ref(),
+                CARGO_PKG_VERSION
+            );
+
+            scope_metrics.metrics.into_iter().for_each(|metric| {
+                assert_eq!(metric.name, self.expected_metric_name);
+
                 match self.expected_instrument_kind {
                     InstrumentKind::Counter | InstrumentKind::UpDownCounter => {
-                        let number = record
-                            .aggregator()
-                            .unwrap()
-                            .as_any()
-                            .downcast_ref::<SumAggregator>()
-                            .unwrap()
-                            .sum()
-                            .unwrap();
-
+                        let sum = metric.data.as_any().downcast_ref::<Sum<T>>().unwrap();
                         assert_eq!(
-                            Ordering::Equal,
-                            number
-                                .partial_cmp(&NumberKind::U64, &self.expected_value)
-                                .unwrap()
+                            self.expected_value,
+                            sum.data_points
+                                .iter()
+                                .map(|data_point| data_point.value)
+                                .sum()
                         );
                     }
                     InstrumentKind::Histogram => {
-                        let histogram = record
-                            .aggregator()
-                            .unwrap()
-                            .as_any()
-                            .downcast_ref::<HistogramAggregator>()
-                            .unwrap()
-                            .histogram()
-                            .unwrap();
-
-                        let counts = histogram.counts();
-                        if dbg!(self.expected_value.to_i64(&self.expected_number_kind)) > 100 {
-                            assert_eq!(counts, &[0.0, 0.0, 1.0]);
-                        } else if self.expected_value.to_i64(&self.expected_number_kind) > 0 {
-                            assert_eq!(counts, &[0.0, 1.0, 0.0]);
-                        } else {
-                            assert_eq!(counts, &[1.0, 0.0, 0.0]);
-                        }
+                        let histogram =
+                            metric.data.as_any().downcast_ref::<Histogram<T>>().unwrap();
+                        let counts = histogram.data_points.first().unwrap().count;
+                        assert!(counts > 0);
```

Comment:
Do I need to be more strict with the histogram metrics assert?

Reply:
could add a sum assertion here too to ensure the value recorded is correct